src/renderers/dom/fiber/ReactDOMFiberComponent.js

```diff
@@ -179,6 +179,19 @@ if (__DEV__) {
           parent.tagName,
         );
     testElement.innerHTML = html;
+
+    if (parent.tagName === 'NOSCRIPT') {
+      // <noscript> content is parsed as text, but only if the browser parses it
+      // together with <noscript> tag itself. So we have to wrap it once more.
```

Comment:
```js
a = document.createElement('noscript');
a.innerHTML = '<b>foobar</b>';
console.log(a.childNodes);
```

Outputs a single TextNode for me in both Chrome and Edge (it does not parse the b-tag), which seems to contradict the comment and the whole reason for this existing... right? Is there more to it than that?

Reply:
Interesting. What you're saying is right for `document.createElement()`, but not for `document.implementation.createHTMLDocument('').createElement()` which is what we're using. I'm not sure why there is a difference (is it because it doesn't have a body or something like this?)

Reply:
There doesn't seem to be a difference between the wrapper and not in this test:
http://jsbin.com/mofexaxaya/2/edit?js,console

What is different?